packages/logging/src/logging-and-telemetry.ts

```diff
@@ -52,6 +53,40 @@ import type {
   MongoshLoggingAndTelemetryArguments,
   MongoshTrackingProperties,
 } from './types';
+import { createHmac } from 'crypto';
+
+/**
+ * @returns A hashed, unique identifier for the running device or `"unknown"` if not known.
+ */
+export async function getDeviceId({
```

Comment:
This could go inside the class but seems clearer to make it independent of it and avoid tying it to it to potentially even expose it on its own if we have some need for that in the future